arducate/src/components/Toolbar.js

```diff
@@ -29,11 +32,82 @@ const Toolbar = () => {
     window.open(url, "_blank");
   };
 
-  const handlePublish = () => {
-    const htmlContent = convertSceneToAR(arObjects);
-    const blob = new Blob([htmlContent], { type: "text/html" });
-    const url = URL.createObjectURL(blob);
-    window.open(url, "_blank");
+  // Handle the publishing action
+  const handlePublish = async () => {
+    if (!generatedUrl) { // Only generate a new URL if one doesn't exist
+      const htmlContent = convertSceneToAR(arObjects);
+      const uniqueId = uuidv4(); // Generate a unique ID for the content
+
+      try {
+        const { data, error } = await supabase
+          .from('ar_content')
+          .insert([
+            { url: uniqueId, html_content: htmlContent }
+          ]);
+  
+        if (error) {
+          throw new Error('Failed to save content: ' + error.message);
+        }
+
+        const baseUrl = window.location.origin; 
+        const arContentUrl = `${baseUrl}/ar-content/${uniqueId}`;        setGeneratedUrl(arContentUrl); // Store the URL
+        sessionStorage.setItem('ar_content_url', arContentUrl); // Persist in sessionStorage
+        window.open(arContentUrl, '_blank'); // Open the generated URL
+      } catch (err) {
+        console.error(err.message);
+      }
+    } else {
+      // If the URL already exists, just open it
+      window.open(generatedUrl, '_blank');
+    }
+  };
+
+  // Handle the sharing action
+  const handleShare = async () => {
+    if (!generatedUrl) { // Only generate a new URL if one doesn't exist
+      const htmlContent = convertSceneToAR(arObjects); // Get the HTML content for the AR scene
+      const uniqueId = uuidv4(); // Generate a unique ID for the content
+      
+      try {
+        const { data, error } = await supabase
+          .from('ar_content')
```

Comment:
probably should rename to something else, what was the schema you went with?

Reply:
what do you want me to name it to? The schema is:

id,uuid, primary key
created_at, timestamp
url, text
html_content, text